### PR TITLE
Migrate IexecERC20Delegate unit tests

### PR DESCRIPTION
**Pull request by jeremyjams**

### CODE REVIEW

File: test/byContract/IexecERC20/IexecERC20.ts [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    TestReceiver,
+    TestReceiver__factory,
+} from '../../../typechain';
+import { NULL } from '../../../utils/constants';
+import { getIexecAccounts } from '../../../utils/poco-tools';
+import { IexecWrapper } from '../../utils/IexecWrapper';
+
+const emptyValue = 0;
+const value = 100;
+const zeroAddress = NULL.ADDRESS;
```
Comment (zguesmi) [inline test/byContract/IexecERC20/IexecERC20.ts:21]: Use `ethers.constants.AddressZero` instead.

File: test/byContract/IexecERC20/IexecERC20.ts [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+    IexecInterfaceNative,
+    IexecInterfaceNative__factory,
+    TestReceiver,
+    TestReceiver__factory,
+} from '../../../typechain';
+import { NULL } from '../../../utils/constants';
+import { getIexecAccounts } from '../../../utils/poco-tools';
+import { IexecWrapper } from '../../utils/IexecWrapper';
+
+const emptyValue = 0;
```
Comment (zguesmi) [inline test/byContract/IexecERC20/IexecERC20.ts:19]: Using the value `0` directly is quite explicit and enough IMO.

File: test/byContract/IexecERC20/IexecERC20.ts (hunk 1 above)
Comment (jeremyjams) [reply test/byContract/IexecERC20/IexecERC20.ts:21]: Lol I used that but changed it for the review!

**Review by gfournieriExec — APPROVED** [review]
Thanks !

File: test/byContract/IexecERC20/IexecERC20.ts [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        iexecPocoAsHolder = iexecPoco.connect(holder);
+        iexecPocoAsSpender = iexecPoco.connect(spender);
+        totalSupplyBeforeFirstDeposit = await iexecPoco.totalSupply();
+        await iexecWrapper.depositInIexecAccount(holder, value);
+    }
+
+    describe('Total supply', () => {
+        it('Should get total supply', async () => {
+            expect(await iexecPoco.totalSupply()).equal(totalSupplyBeforeFirstDeposit.add(value));
+        });
```
Comment (zguesmi) [inline test/byContract/IexecERC20/IexecERC20.ts:52]: This is also tested in `IexecAccessors`. Let's choose one of them.

**Review by zguesmi — APPROVED** [review]

File: test/byContract/IexecERC20/IexecERC20.ts (hunk 3 above)
Comment (jeremyjams) [reply test/byContract/IexecERC20/IexecERC20.ts:52]: I would say both tests makes sense, here for instance testing https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/IERC20.sol

**Pull request merged**